diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index 10939a0bd..170326625 100755
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -174,7 +174,7 @@ def is_windows():
     return sys.platform.startswith("win")
 
 def is_ubuntu_1604():
-    return platform.linux_distribution()[0] == 'Ubuntu' and platform.linux_distribution()[1] == '16.04'
+    return False
 
 def get_config_build_dir(build_dir, config):
     # build directory per configuration
